lib/src/mobile_scanner_preview.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+
+/// A widget showing a live camera preview.
+class CameraPreview extends StatelessWidget {
+  /// Creates a preview widget for the given camera controller.
+  const CameraPreview(this.controller, {super.key});
+
+  /// The controller for the camera that the preview is shown for.
+  final MobileScannerController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    if (!controller.value.isInitialized) {
+      return const SizedBox();
+    }
+
+    return ValueListenableBuilder<MobileScannerState>(
+      valueListenable: controller,
+      builder: (BuildContext context, MobileScannerState value, Widget? child) {
+        return SizedBox.fromSize(
+          size:
+              controller.value.deviceOrientation.isLandscape
```

Comment:
You can use the value directly here

```suggestion
              value.deviceOrientation.isLandscape
```